thunder/transforms/cudagraph.py

```diff
@@ -100,44 +111,79 @@ def __call__(self, *args):
         graph.replay()
         return static_outputs
 
+    def make_python_callable_from_symbols(
+        self,
+        fn_name: str,
+        bsyms: list[BoundSymbol],
+        inputs: list[Proxy],
+        outputs: list[Proxy],
+    ) -> Callable:
 
-def make_callable(
-    fn_name: str,
-    bsyms: list[BoundSymbol],
-    inputs: list[Proxy],
-    outputs: list[Proxy],
-) -> Callable:
-    from inspect import Parameter, Signature
+        from inspect import Parameter, Signature
 
-    region_fn_params = (
-        Parameter(getattr(param, "name", f"arg{i}"), Parameter.POSITIONAL_ONLY) for i, param in enumerate(inputs)
-    )
+        region_fn_params = (
+            Parameter(getattr(param, "name", f"arg{i}"), Parameter.POSITIONAL_ONLY) for i, param in enumerate(inputs)
+        )
 
-    region_fn_signature = Signature(region_fn_params)
+        region_fn_signature = Signature(region_fn_params)
 
-    def region_fn():
-        pass
+        def region_fn():
+            pass
 
-    region_fn.__signature__ = region_fn_signature
-    region_fn.__name__ = fn_name
+        region_fn.__signature__ = region_fn_signature
+        region_fn.__name__ = fn_name
 
-    region_trace = TraceCtx(region_fn)
-    region_trace.bound_symbols = bsyms
-    region_trace.args = inputs
-    region_trace.kwargs = {}
-    region_trace.bound_symbols.append(prims.python_return.bind(outputs, output=()))
+        region_trace = TraceCtx(region_fn)
+        region_trace.bound_symbols = bsyms
+        region_trace.args = inputs
+        region_trace.kwargs = {}
+        region_trace.bound_symbols.append(prims.python_return.bind(outputs, output=()))
+        return region_trace.python_callable()
 
-    return region_trace.python_callable()
+    def make_cuda_graph_callable_from_symbols(
+        self,
+        fn_name: str,
+        bsyms: list[BoundSymbol],
+        inputs: list[Proxy],
+        outputs: list[Proxy],
+        num_static_inputs: int | None = None,
+    ) -> Callable:
+
+        if num_static_inputs is not None:
+            static_inputs_mask = (True,) * num_static_inputs + (False,) * (len(inputs) - num_static_inputs)
+        else:
+            static_inputs_mask = None
+
+        x_fn_name = f"{fn_name}_{self.name_counter}"
+        self.name_counter += 1
+
+        self.python_callables[x_fn_name] = (
+            self.make_python_callable_from_symbols(fn_name, bsyms, inputs, outputs),
+            static_inputs_mask,
+        )
+        self.trace_symbols[x_fn_name] = (bsyms, inputs, outputs)
```

Comment:
How can I access these dictionaries in the regular user script? How should developers get to know that this information is saved and available for inspection?

Reply:
Currently, you have to look at the source. I'm not yet sure that this is the exact information we want to keep, we should refine this.

Reply:
so in addition to everything else the `x_fn_name` is a bit stupid, the trick was that the transform is applied to forward and backward. If we are OK with cudagraph regions not starting from 1 in the backward, I would just generate the name in the runner and return it.
This would allow a link from trace->cache name here easily. Is that what you have in mind?